client/services/PlanningStoreService.js

```diff
@@ -0,0 +1,102 @@
+import { isNil } from 'lodash'
+import { createStore } from '../utils'
+import * as actions from '../actions'
+
+PlanningStoreService.$inject = [
+    '$rootScope',
+    'api',
+    'config',
+    '$location',
+    '$timeout',
+    'vocabularies',
+    'superdesk',
+    'upload',
+    'notify',
+    'privileges',
+    'notifyConnectionService',
+    'userList',
+    'desks',
+    'metadata',
+    'session',
+    'deployConfig',
+    'gettextCatalog',
+]
+export function PlanningStoreService(
+    $rootScope,
+    api,
+    config,
+    $location,
+    $timeout,
+    vocabularies,
+    superdesk,
+    upload,
+    notify,
+    privileges,
+    notifyConnectionService,
+    userList,
+    desks,
+    metadata,
+    session,
+    deployConfig,
+    gettextCatalog
+) {
+    let self = this
+
+    self.store = null
```

Comment:
The service is implemented as a Singleton Instance, created upon first use. This reduces the overhead required to create the Redux store, especially when used in the Superdesk Authoring views (such as creating a Planning item from a story).

The following are the current reducers and when they're loaded:

| Permanent  | Ad hoc |
| ------------- | ------------- |
| desks | agenda |
| genres | assignment |
| ingest_providers | events |
| modal | forms |
| privileges | formsProfile |
| session | planning |
| subjects |  |
| urgency |  |
| users |  |
| vocabularies |  |

The ad hoc reducers are set to `null` when `actions.resetStore` is dispatched, and set to `initialState` when `actions.initStore` is dispatched (ready for use by other actions and components).

The other way would be to create the Redux store from scratch every time, and populate with all the reducers (with API calls for almost all reducers).

Let me know which you would prefer and I will make the changes if necessary.